src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -448,12 +375,12 @@ private int FindLatestEventHandlerIdInChain(int eventHandlerId)
 
         private void EnsureSynchronizationContext()
         {
-            // When the IDispatcher is a synchronization context
+            // When the Dispatcher is a synchronization context
             // Render operations are not thread-safe, so they need to be serialized.
             // Plus, any other logic that mutates state accessed during rendering also
             // needs not to run concurrently with rendering so should be dispatched to
             // the renderer's sync context.
-            if (_dispatcher is SynchronizationContext synchronizationContext && SynchronizationContext.Current != synchronizationContext)
+            if (!Dispatcher.CheckAccess())
             {
                 throw new InvalidOperationException(
                     "The current thread is not associated with the renderer's synchronization context. " +
```

Comment:
The exception message now seems a bit off, because `Renderer` isn't meant to be aware of the concept of a synchronization context. All it knows is that `Dispatcher.CheckAccess()` returned `false`; it doesn't know why.

I guess if we wanted to relayer this we could have a `Dispatcher.AssertAccess` method that throws an exception whose message could vary to match whatever rules the dispatcher is using to check access.

I don't consider this particularly vital to clean up right now - we'll always have the option to do it in the future. So I'm sure you can decide whether or not it's something that bothers you.

Reply:
I polished this error message to be clearer.